src/commitment.rs

```diff
@@ -124,6 +85,136 @@ impl<C: CurveGroup> HashCommitmentResult<C> {
             ResultValue::Scalar(out)
         });
 
-        HashCommitmentResult { value, blinder, commitment: comm }
+        HashCommitmentResult { values, blinder, commitment: comm }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use futures::future;
+    use itertools::Itertools;
+    use rand::thread_rng;
+
+    use crate::{
+        algebra::{CurvePoint, Scalar, ToBytes},
+        commitment::{HashCommitment, HashCommitmentResult},
+        test_helpers::{execute_mock_mpc, TestCurve},
+        ResultValue,
+    };
+
+    /// Verify a commitment to a value
+    async fn verify_comm<T: From<ResultValue<TestCurve>> + Unpin + ToBytes>(
+        comm: HashCommitmentResult<TestCurve, T>,
+    ) -> bool {
+        let comm_res = HashCommitment {
+            blinder: comm.blinder,
+            values: future::join_all(comm.values).await,
+            commitment: comm.commitment.await,
+        };
```

Comment:
is it worth it to `impl Future for HashCommitmentResult` for this to be more ergonomic (if we do this outside of testing code)?

Reply:
We don't do this outside testing code, they're never awaited locally, just shipped off to the recipient